src/circuit/qc_keyagg.rs

```diff
@@ -1,26 +1,71 @@
 //! Circuit implementation of stake key aggregation for quorum certificates verification.
-use ark_ec::{
-    short_weierstrass::{Projective, SWCurveConfig},
-    CurveGroup,
-};
+
 use ark_ff::PrimeField;
-use ark_std::{format, vec, Zero};
-use jf_primitives::{circuit::rescue::RescueNativeGadget, rescue::RescueParameter};
+use ark_std::{format, vec, vec::Vec};
+use jf_primitives::{
+    circuit::rescue::RescueNativeGadget,
+    rescue::{sponge::RescueCRHF, RescueParameter},
+};
 use jf_relation::{
     errors::CircuitError,
     gadgets::{
-        ecc::{non_native::EmulatedPointVariable, Point, SWToTEConParam},
-        EmulationConfig,
+        ecc::{
+            emulated::{EmulatedSWPointVariable, EmulatedTEPointVariable, SWPoint},
+            TEPoint,
+        },
+        EmulationConfig, SerializableEmulatedStruct,
     },
     BoolVar, Circuit, PlonkCircuit, Variable,
 };
 
-#[derive(Debug, Clone)]
-/// Stake public key variable
-/// Wrap EmulatedPointVariable because we need to simulate curve operations over the same curve's scalar field
-pub struct VerKeyVar<E: PrimeField>(pub EmulatedPointVariable<E>);
+pub fn compute_stake_table_hash<F: RescueParameter, T: SerializableEmulatedStruct<F>>(
```

Comment:
Add some comment for public APIs.

Reply:
Done in [8697221](https://github.com/EspressoSystems/hotshot-primitives/pull/74/commits/869722169145e40b275b9d16e50284a3868e2a60)